agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4416,27 +4416,60 @@ bool slave_thread::autoconfig_wsc_parse_m2_encrypted_settings(
     return true;
 }
 
+/**
+ * @brief Parse AP-Autoconfiguration WSC which should include one AP Radio Identifier
+ *        TLV and one or more WSC TLV containing M2
+ * @param sd socket descriptor
+ * @param cmdu_rx received CMDU containing M2
+ * @return true on success
+ * @return false on failure
+ */
 bool slave_thread::handle_autoconfiguration_wsc(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
 {
-    // Check if this is a M1 message that we sent to the controller, which was just looped back.
-    // The M1 and M2 messages are both of CMDU type AP_Autoconfiguration_WSC. Thus,
-    // when we send the M2 to the local agent, it will be published back on the local bus because
-    // the destination is our AL-MAC, and the controller does listen to this CMDU.
-    // Ideally, we should use the message type attribute from the WSC payload to distinguish.
-    // Unfortunately, that is a bit complicated with the current tlv parser. However, there is another
-    // way to distinguish them: the M1 message has the AP_Radio_Basic_Capabilities TLV,
-    // while the M2 has the AP_Radio_Identifier TLV.
-    // If this is a looped back M1 CMDU, we can treat is as handled successfully.
-    if (cmdu_rx.getNextTlvType() == int(wfa_map::eTlvTypeMap::TLV_AP_RADIO_BASIC_CAPABILITIES)) {
-        LOG(DEBUG) << "looped back M1 CMDU";
-        return true;
-    }
+    LOG(DEBUG) << "Received AP_AUTOCONFIGURATION_WSC_MESSAGE";
+
+    std::shared_ptr<wfa_map::tlvApRadioIdentifier> ruid = nullptr;
+    std::vector<std::shared_ptr<ieee1905_1::tlvWscM2>> m2_list;
+    bool intel_agent = false;
+    int type         = cmdu_rx.getNextTlvType();
+    do {
+        if (type == int(wfa_map::eTlvTypeMap::TLV_AP_RADIO_IDENTIFIER)) {
```

Comment:
I would prefer a switch - which is difficult here because of the (IMO unnecessary) distinction between ieee1905_1::eTlvType and wfa_map::eTlvTypeMap.

Reply:
indeed it is more difficult, so lets stick with if / else